app/liquid_tags/cta_tag.rb

```diff
@@ -0,0 +1,48 @@
+class CtaTag < Liquid::Block
+  include ActionView::Helpers::SanitizeHelper
+
+  PARTIAL = "liquids/cta".freeze
+  STYLE_OPTIONS = %w[branded].freeze
+  WIDTH_OPTIONS = %w[inline block].freeze
+
+  def initialize(_tag_name, options, _parse_context)
+    super
+
+    options = strip_tags(options.strip).split
+
+    @link = link(options)
```

Comment:
This is where I expect that we may have some discussions. Captured below are some of the requirements for the CTA:

**1. We're hoping to be able to possibly style the CTA's differently in the future. Not necessarily a requirement right now.** 

Right now we use `ltag_cta--branded` as our default. I've added the ability to add an option `branded` to our CTA but it defaults to `branded` anyways if an option is not provided. This sets the scene for future branding whilst not requiring an option to be set now. 

```
{% cta https://my-link.com/link <style> %}
Click my awesome button!
{% endcta %}
```

If style is not set, we will just use `branded`.

This was simple to incorporate hence I included it, however I can build this in whilst not exposing the `style` option to the user for now if we'd prefer to rather incorporate this in the future. 

**2. We need a full width CTA in the sidebar and other locations are just inline block.** 

There are two options on how implement this:

- The first option is the way I've done it now by adding an option to the CTA. The options will be either `block` or `inline` and if one is not provided it will default to `inline`. If the option term `block` is not intuitive, then I can map it to `full-width`.
- The other option is to put unique classnames on the sidebar billboards vs the feed/individual article billboard and style the CTA appropriately based on that. This means that we'll need to track which classname we'll need to add to which placement areas. We'll need to be conscious of adding these classnames as we add more placement areas. This feels slightly less flexible and hence I went with the first option. 

```
{% cta https://my-link.com/link <width> %}
Click my awesome button!
{% endcta %}
```

For example:
```
{% cta https://my-link.com/link block %}
Click my awesome button!
{% endcta %}

{% cta https://my-link.com/link inline %}
Click my awesome button!
{% endcta %}
```

If we do not provide an option it will default to inline. 

I'm open to more thoughts on this? 

**3. We want to be able to default to left alignment for the CTA but at times mark it as center.**

I haven't implemented this as yet but since this seems like a configuration, I think we'd need to add it similarly to the above: 

```
{% cta https://my-link.com/link <alignment> %}
Click my awesome button!
{% endcta %}
```

For example:
```
{% cta https://my-link.com/link  %}
Click my awesome button!
{% endcta %}
```

For example:
```
{% cta https://my-link.com/link center|right|left %}
Click my awesome button!
{% endcta %}
```

We could eventually even add right alignment in this way if we wanted to. 

**4. In mobile we will have full width CTA for all billboard units.**

In this case I've used css to achieve this.  

Desktop:
<img width="1037" alt="Screenshot 2023-05-08 at 13 48 39" src="https://user-images.githubusercontent.com/2786819/236816331-ea77b28f-48c5-42c5-8632-e39bac269092.png">

Mobile: 
<img width="445" alt="Screenshot 2023-05-08 at 13 48 51" src="https://user-images.githubusercontent.com/2786819/236816356-8f36dbec-576d-484b-b32c-609a38438af8.png">


**5. Other:** 
- Font size and weight are inherited within the context of the page so that it adheres to the requirements
![image](https://user-images.githubusercontent.com/2786819/236835577-fdde36c9-ee59-4ca4-ad86-9e0d9c96f106.png)

![image](https://user-images.githubusercontent.com/2786819/236835600-196e4f0f-8f2b-4ce2-83fa-2df403c67047.png)


Here's a page of some examples: 

![screencapture-localhost-3000-2023-05-08-15_30_36](https://user-images.githubusercontent.com/2786819/236837307-13747ed0-6275-4408-8140-23f3a42d3f50.png)

cc @anujbhavsar96 @benhalpern

Reply:
> The other option is to put unique classnames on the sidebar billboards vs the feed/individual article billboard and style the CTA appropriately based on that. This means that we'll need to track which classname we'll need to add to which placement areas. We'll need to be conscious of adding these classnames as we add more placement areas. This feels slightly less flexible and hence I went with the first option.

I think at the end of the day, we should initially dictate this strictly by _context_, and actually not inject any classname into the embed's HTML for this.

So that is to say — I think the only option could be stuff that dictates primary/secondary/etc. and everything else is context.

And if we wanted to add additional options _later_ we could.

But with not injecting classnames, what I mean is that we can use the wrapper class + media queries to figure out context.

That might add a few extra lines to this css file, and make it a little hairy to handle edge cases, but this will actually simplify our capacity to fix unhandled edge cases later, or change up the look without having to go in and re-save classes.

So that is to say — I think we can define the default styles for the CTA, and the topline options, but then add some custom rules based on some wrappers, such as

```
.crayons-article-sticky {
  display: block
}
```

This just happens to be the pattern we've established to deal with different scenarios for embeds and billboard styling in general, so I think we need to lean into it.

If you look at the bottom of `app/assets/stylesheets/ltags/LinkTag.scss` you'll see some special styles for the `crayons-article-sticky ` (i.e. billboard sidebar).

While I don't think this is the cleanest pattern, it is a pretty highly refactorable one, all else equal. And it's now the established one.

It maximizes our chance to keep this PR reasonable in size and stamp out any possible style edge cases as a self-contained task later.

Reply:
Still working on (1) but I've fixed (2)

Mobile: 
<img width="460" alt="Screenshot 2023-05-09 at 11 55 24" src="https://user-images.githubusercontent.com/2786819/237061902-6d36b1a2-be8f-42e7-9095-4878ffbce782.png">

whilst keeping it the same for Desktop: 
<img width="492" alt="Screenshot 2023-05-09 at 11 56 42" src="https://github.com/forem/forem/assets/2786819/a6c96f45-8180-43ac-bda4-91ea0ed8998c">

Reply:
This looks great. Thank you ✨